be/src/exprs/cast_expr.cpp

```diff
@@ -1703,6 +1703,36 @@ StatusOr<std::unique_ptr<Expr>> VectorizedCastExprFactory::create_cast_expr(Obje
                                                                             const TypeDescriptor& from_type,
                                                                             const TypeDescriptor& to_type,
                                                                             bool allow_throw_exception) {
+    if (to_type.is_string_type() && from_type.is_array_type()) {
+        LOG(INFO) << "Cast from array to varchar. ";
+        TypeDescriptor array_field_type_cast_to = TypeDescriptor::from_thrift(node.type);
+        TypeDescriptor array_field_type_cast_from = TypeDescriptor::from_thrift(node.child_type_desc);
+
+        while (from_type.is_array_type() && to_type.is_string_type()) {
```

Comment:
why need while?

Reply:
if the input array is two-dimensional, for safety we'd better loop using while.

Reply:
but you didn't change from_type or to_type in this loop…is there something wrong？

Reply:
> but you didn't change from_type or to_type in this loop…is there something wrong？

@MichaelYun7 how about this question?